services/merak-topo/grpc/service/service.go

```diff
@@ -72,20 +72,22 @@ func (s *Server) TopologyHandler(ctx context.Context, in *pb.InternalTopologyInf
 			log.Printf("return host node %v", returnMessage.Hosts)
 		}
 
-	case common_pb.OperationType_CREATE:
+	case pb_common.OperationType_CREATE:
 
 		/// save the protobuf hostinfo --- with each created topology
 
 		aca_num := in.Config.GetNumberOfVhosts()
 		rack_num := in.Config.GetNumberOfRacks()
 		aca_per_rack := in.Config.GetVhostPerRack()
 		data_plane_cidr := in.Config.GetDataPlaneCidr()
-		cgw_num := in.Config.GetNumberOfGateways()
+		// cgw_num := in.Config.GetNumberOfGateways()
+		cgw_num := 0
 		topo_id := in.Config.GetTopologyId()
 
-		if data_plane_cidr == "" || aca_num == 0 || aca_per_rack == 0 || rack_num == 0 || cgw_num == 0 {
+		// if data_plane_cidr == "" || aca_num == 0 || aca_per_rack == 0 || rack_num == 0 || cgw_num == 0 {
```

Comment:
@zzengcs Please remove it if you don't need them.

Reply:
I think the better way to handle the gateway input is that: we need to check if the gateway is required by user, if user indeed require the gateway, we still need to create the gateway pods for them. We can just ignore the gateway for now, but please keep-in-mind how to design a good way to handle the gateway requirement from user.

Reply:
Resolved by adding comment notes.